src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -282,18 +281,25 @@ class MySimulation : public SimulationHelper<simDim>
 
         size_t freeGpuMem(0);
         Environment<>::get().EnvMemoryInfo().getMemoryInfo(&freeGpuMem);
-        freeGpuMem -= totalFreeGpuMemory;
+        if(freeGpuMem < reservedGpuMemorySize)
+        {
+            PMacc::log< picLog::MEMORY > ("%1% MiB free memory < %2% MiB required reserved memory")
+                % (freeGpuMem / 1024 / 1024) % (reservedGpuMemorySize / 1024 / 1024) ;
+            throw std::runtime_error("Cannot reserve enough memory");
```

Comment:
Yes you are right, but if this error is triggered on a big 18k gpu machine it is very expensive to rerun the simulation and might be the error is not triggered again.